ios/ImagePickerManager.m

```diff
@@ -203,14 +203,27 @@ - (void)launchImagePicker:(RNImagePickerTarget)target
         }];
     }
     else { // RNImagePickerTargetLibrarySingleImage
-        [self checkPhotosPermissions:^(BOOL granted) {
-            if (!granted) {
-                self.callback(@[@{@"error": @"Photo library permissions not granted"}]);
-                return;
-            }
-
+        // Wrapper for permission check and showing picker
+        void (^showPickerViewControllerIfPermitted)() = ^() {
+            [self checkPhotosPermissions:^(BOOL granted) {
+                if (!granted) {
+                    self.callback(@[@{@"error": @"Photo library permissions not granted"}]);
+                    return;
+                }
+                
+                showPickerViewController();
+            }];
+        };
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 110000
```

Comment:
I included this macro check as I don't want to assume someone will only be using this on Xcode 9 and up.

Reply:
I think it would make more sense to include this logic within `showPickerViewControllerIfPermitted`

Reply:
Were you envisioning something along this?
```objective-c
    // Wrapper for permission check and showing picker
    void (^showPickerViewControllerIfPermitted)() = ^() {
#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 110000
        if (@available(iOS 11.0, *)) {
            // iOS 11 no longer requires permission to access single image
            showPickerViewController();
        } else {
            [self checkPhotosPermissions:^(BOOL granted) {
                if (!granted) {
                    self.callback(@[@{@"error": @"Photo library permissions not granted"}]);
                    return;
                }
                
                showPickerViewController();
            }];
        }
#else
        [self checkPhotosPermissions:^(BOOL granted) {
            if (!granted) {
                self.callback(@[@{@"error": @"Photo library permissions not granted"}]);
                return;
            }
            
            showPickerViewController();
        }];
#endif
    };

    showPickerViewControllerIfPermitted();
```

Reply:
Yes, it seems cleaner since all the necessary permission checks are encapsulated in `showPickerViewControllerIfPermitted`. Looks great 👍

Reply:
Sure, I can dig it. I don't like the repetition but I feel like that is unavoidable. I'll try to update this then later tonight.

Off topic: Do you know if this repo is still being maintained?